Applications/Scale/test/testScale.cpp

```diff
@@ -460,4 +470,329 @@ void scalePhysicalOffsetFrames()
+        model->addBody(body1);
+        OpenSim::Body* body2 = new OpenSim::Body("body2", 1, Vec3(0), Inertia(0));
+        model->addBody(body2);
+
+        SpatialTransform transform;
+        transform[3].setCoordinateNames(OpenSim::Array<std::string>("x", 1, 1));
+        transform[3].setFunction(new LinearFunction(1.2, 3.4));
+
+        // Evaluate LinearFunction before scaling and compare after scaling.
+        const Vector xA = Vector(1, 0.);
+        const double yA = transform[3].getFunction().calcValue(xA);
+        const Vector xB = Vector(1, 1.618);
+        const double yB = transform[3].getFunction().calcValue(xB);
+
+        CustomJoint* custom1 =
+            new CustomJoint("custom1", model->getGround(), *body1, transform);
+        model->addJoint(custom1);
+        CustomJoint* custom2 =
+            new CustomJoint("custom2", *body2, model->getGround(), transform);
+        model->addJoint(custom2);
+
+        // Scale the model.
+        State& s = model->initSystem();
+        model->scale(s, scaleSet, false);
+
+        // Transform of custom1 should not have changed.
+        const OpenSim::Function& fn1 =
+            custom1->getSpatialTransform()[3].getFunction();
+        ASSERT_EQUAL(fn1.calcValue(xA), yA, SimTK::SignificantReal,
+            __FILE__, __LINE__, "CustomJoint 'custom1' was incorrectly scaled.");
+        ASSERT_EQUAL(fn1.calcValue(xB), yB, SimTK::SignificantReal,
+            __FILE__, __LINE__, "CustomJoint 'custom1' was incorrectly scaled.");
+
+        // Transform of custom2 should have been scaled using the scale factors
+        // corresponding to body2.
+        const OpenSim::Function& fn2 =
+            custom2->getSpatialTransform()[3].getFunction();
+        ASSERT_EQUAL(fn2.calcValue(xA), yA * scaleFactors[0],
+            SimTK::SignificantReal, __FILE__, __LINE__,
+            "CustomJoint 'custom2' was incorrectly scaled.");
+        ASSERT_EQUAL(fn2.calcValue(xB), yB * scaleFactors[0],
+            SimTK::SignificantReal, __FILE__, __LINE__,
+            "CustomJoint 'custom2' was incorrectly scaled.");
+    }
+
+    // Test CoordinateCouplerConstraint scaling.
+    {
+        cout << "- CoordinateCouplerConstraint" << endl;
+
+        Model* model = new Model();
+        OpenSim::Body* body1 = new OpenSim::Body("body1", 1, Vec3(0), Inertia(0));
+        model->addBody(body1);
+        OpenSim::Body* body2 = new OpenSim::Body("body2", 1, Vec3(0), Inertia(0));
+        model->addBody(body2);
+
+        // Attach bodies to Ground with slider joints.
+        SliderJoint* slider1 =
+            new SliderJoint("slider1", model->getGround(), *body1);
+        slider1->updCoordinate().setName("indepCoord");
+        model->addJoint(slider1);
+
+        SliderJoint* slider2 =
+            new SliderJoint("slider2", *body2, model->getGround());
+        slider2->updCoordinate().setName("depCoord");
+        model->addJoint(slider2);
```

Comment:
I was confused when I saw this because I thought the test might be similar to the tests above (making sure the scaling behavior differs depending on the parent frame). Perhaps construct this joint as `SliderJoint(..., *body1, *body2)`?

Reply:
Ok, sure.